apps/etterlatte-beregning/src/main/kotlin/avkorting/Avkorting.kt

```diff
@@ -151,31 +152,49 @@ data class Avkorting(
         return oppdatertMedNyInntekt.beregnAvkortingRevurdering(beregning, sanksjoner)
     }
 
+    private fun haandterOpphoerDato(
+        opphoerFom: YearMonth? = null,
+        nyttGrunnlag: AvkortingGrunnlagLagreDto,
+    ): YearMonth? =
+        // opphoerFom kan maks være ett år frem
+        opphoerFom?.let {
+            if (it.year.minus(nyttGrunnlag.fom.year) == 1) {
+                YearMonth.of(nyttGrunnlag.fom.year, Month.DECEMBER)
+            } else if (it.year == nyttGrunnlag.fom.year) {
+                it
+            } else {
+                throw OpphoerErTilbakeITid(
+                    "Opphøer er tilbake i tid, opphør: $opphoerFom nyttgrunnlag er fra: ${nyttGrunnlag.fom} id: ${nyttGrunnlag.id}",
+                )
+            }
+        }
+
     fun oppdaterMedInntektsgrunnlag(
         nyttGrunnlag: AvkortingGrunnlagLagreDto,
         bruker: BrukerTokenInfo,
         opphoerFom: YearMonth? = null,
         aldersovergang: YearMonth? = null,
     ): Avkorting {
+        val nyttOpphoerFom = haandterOpphoerDato(opphoerFom, nyttGrunnlag)
```

Comment:
nitpick da men siden det nå kan være tom desember og for det året ikke opphør så bør kanskje navnet reflektere det. Holder med tom elns.

Reply:
Skjønner ikke helt hva du mener med "tom" desember?
Betyr det at det her kun vil bli en avkorting hvis det skjer en revurdering?